backend/dataall/cdkproxy/stacks/pivot_role.py

```diff
@@ -0,0 +1,822 @@
+                        'redshift:resumeCluster',
+                    ],
+                    resources=['*'],
+                    conditions={'StringEquals': {'aws:ResourceTag/dataall': 'true'}},
+                ),
+                iam.PolicyStatement(
+                    sid='RedshiftRead',
+                    effect=iam.Effect.ALLOW,
+                    actions=[
+                        'redshift:DescribeClusters',
+                        'redshift:CreateTags',
+                        'redshift:DescribeClusterSubnetGroups',
+                    ],
+                    resources=['*'],
+                ),
+                iam.PolicyStatement(
+                    sid='RedshiftCreds',
+                    effect=iam.Effect.ALLOW,
+                    actions=['redshift:GetClusterCredentials'],
+                    resources=[
+                        f'arn:aws:redshift:*:{self.account}:dbgroup:*/*',
+                        f'arn:aws:redshift:*:{self.account}:dbname:*/*',
+                        f'arn:aws:redshift:*:{self.account}:dbuser:*/*',
+                    ],
+                ),
+                iam.PolicyStatement(
+                    sid='AllowRedshiftSubnet',
+                    effect=iam.Effect.ALLOW,
+                    actions=['redshift:CreateClusterSubnetGroup'],
+                    resources=['*'],
+                ),
+                iam.PolicyStatement(
+                    sid='AllowRedshiftDataApi',
+                    effect=iam.Effect.ALLOW,
+                    actions=[
+                        'redshift-data:ListTables',
+                        'redshift-data:GetStatementResult',
+                        'redshift-data:CancelStatement',
+                        'redshift-data:ListSchemas',
+                        'redshift-data:ExecuteStatement',
+                        'redshift-data:ListStatements',
+                        'redshift-data:ListDatabases',
+                        'redshift-data:DescribeStatement',
+                    ],
+                    resources=['*'],
+                ),
+                # EC2
+                iam.PolicyStatement(
+                    sid='EC2SG',
+                    effect=iam.Effect.ALLOW,
+                    actions=['ec2:CreateSecurityGroup', 'ec2:CreateNetworkInterface', 'ec2:Describe*'],
+                    resources=['*'],
+                ),
+                iam.PolicyStatement(
+                    sid='TagsforENI',
+                    effect=iam.Effect.ALLOW,
+                    actions=['ec2:CreateTags', 'ec2:DeleteTags'],
+                    resources=[f'arn:aws:ec2:*:{self.account}:network-interface/*'],
+                ),
+                iam.PolicyStatement(
+                    sid='DeleteENI',
+                    effect=iam.Effect.ALLOW,
+                    actions=['ec2:DeleteNetworkInterface'],
+                    resources=['*'],
+                    conditions={'StringEquals': {'aws:ResourceTag/dataall': 'True'}},
```

Comment:
this is incorrect, if we attach pivot role to the lambda function or the crawler ENIs created by the lambda or glue service doesn't contain this dataall tag, as a result ENIs are not released which causes abnormal consumption of IP addresses in the VPC